04-vue-cli/02-TheToaster/components/TheToast.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div :class="['toast', data.className]">
+    <UiIcon class="toast__icon" :icon="data.icon" />
+    <span>{{ text }}</span>
+  </div>
+</template>
+
+<script lang="ts">
+import UiIcon from './UiIcon.vue';
+import { defineComponent } from 'vue';
+
+export enum EToastKind {
+  SUCCESS = 'SUCCESS',
+  ERROR = 'ERROR',
+}
```

Comment:
Very discussible, but it's better to use typescript only for typing. It makes it closer to JS as language outside static type analysis, is easier to compile. ENUM is not just typing but also a 
new syntax expression. Also, ENUMs have many hidden drawbacks.

A simple alternative here is a const object.

Reply:
If we use a const object here, we have to use a `string` instead of `EToastKind` in `TToasterData`:
`type TToasterData = { toasts: { kind: EToastKind; text: string }[] };`
It's decrease a profit of typings.

Reply:
> If we use a const object here, we have to use a string instead of EToastKind in TToasterData

We don't. We still can use values from the const. [Example](https://www.typescriptlang.org/play#code/MYewdgzgLgBAomArgWwIIVWA8gIwFYCmwsAvDAN4BQMNMAygKoDCTcddAXDAOSMtt1uAGmq04AJXFZxXbhKnjhlAL4wAhhBihIUSpSgBPAA4EYAFRAbSFUTQDWASzAATLoZMgAZvCRoM2fCIoAG07AgMvGHcCSIQUdExcQmIAXRFaKIIADyguaAAnJwBzEWU9AHpymBB8mAB3BygAC3UYJoIAGxN8vWiYADU1DsQCCAAeMxhsqAIXTRBA4gA+GDIzUPDIsxTe41MLKwAmVZsMxxcuQeHRsejY3wSA5Kgl9NoZnLyoQrASlSA)

Reply:
There is very verbose syntax...
What is the reason for using a const object instead of ENUM?